src/Dialect/ONNX/ONNXOps/Math/ElementwiseUnary.cpp

```diff
@@ -326,8 +326,8 @@ LogicalResult ONNXIsInfOp::inferShapes(
 
 LogicalResult ONNXIsNaNOp::inferShapes(
     std::function<void(Region &)> doShapeInference) {
-  IntegerType i1Type = IntegerType::get(getContext(), 1, IntegerType::Signless);
-  return inferShapeForUnaryOps(getOperation(), i1Type);
+  return inferShapeForUnaryOps(this->getOperation(),
+      this->getResult().getType().cast<ShapedType>().getElementType());
```

Comment:
Can it be guaranteed the output type is ShapedType?

Reply:
I can change It back to be safe... no guarantee